samples/basic/rgb_gradient/src/RGB_states.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+/* 0-255 based on color codes */
+typedef struct RGB_color {
```

Comment:
Please avoid typedefs as per the coding style guidelines.

Reply:
I guess this is not directly mentioned in the zephyr style guide but comes from the linux style guide which zephyr suggests to follow, so I'll change it